lib/ldap.c

```diff
@@ -182,6 +182,41 @@ const struct Curl_handler Curl_handler_ldaps = {
 };
 #endif
 
+#ifdef USE_WIN32_LDAP
+static int Win_ldap_bind(struct connectdata *conn, LDAP *server, TCHAR *user, TCHAR *passwd)
+{
+    int rc = 0;
+    ULONG method = LDAP_AUTH_NEGOTIATE;
+    SEC_WINNT_AUTH_IDENTITY cred = { 0 };
+    PCHAR inuser = NULL;
+    PCHAR inpass = NULL;
+
+    if (user != NULL && passwd != NULL) {
+        cred.User = user;
+        cred.UserLength = strlen(user);
+        cred.Password = passwd;
+        cred.PasswordLength = strlen(passwd);
+        cred.Flags = SEC_WINNT_AUTH_IDENTITY_ANSI;
```

Comment:
*Jay>* I don't know if LDAP_UNICODE is tied to UNICODE but I'd guess so.

Almost. From <WinLdap.h> in the Win-10 SDK:
```
//
//  The #define LDAP_UNICODE controls if we map the undecorated calls to
//  their unicode counterparts or just leave them defined as the normal
//  single byte entry points.
//
//  If you want to write a UNICODE enabled application, you'd normally
//  just have UNICODE defined and then we'll default to using all LDAP
//  Unicode calls.
//

#ifndef LDAP_UNICODE
#ifdef UNICODE
#define LDAP_UNICODE 1
#else
#define LDAP_UNICODE 0
#endif
#endif
```

Hence `libcurl` (or `cur`l) could do:
```
#define LDAP_UNICODE 0
```

regardless of `UNICODE =[0|1]`. But I fail to see the point of that.

*Jay>* ... Apparently it's possible to build curl with UNICODE, and that may enable some features otherwise not 
*Jay>* available. This is why some functions use TCHARs.

AFAICR, a `libcurl` built with `UNICODE=1` links fine. But a `curl` with the same has problems.